userbenchmark/ddp_experiments/__init__.py

```diff
@@ -489,6 +469,222 @@ def get_backend_name(model_args):
     print(job.results())
 
 
+def apply_fsdp_hf_T5_large(model, trainer):
+    from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+    from torch.distributed.fsdp.wrap import transformer_auto_wrap_policy
+    import functools
+    assert trainer == "fsdp"
+    local_rank = int(os.getenv("LOCAL_RANK", -1))
+    from transformers.models.t5.modeling_t5 import T5Block
+    fsdp_model = FSDP(
+        model,
+        auto_wrap_policy=functools.partial(transformer_auto_wrap_policy, transformer_layer_cls=(T5Block,)),
+        device_id = torch.cuda.current_device(),
+        use_orig_params=True,
+    )
+    print(fsdp_model, "local_rank", local_rank)
+    return fsdp_model
+
+def apply_fsdp_hf_GPT2_large(model, trainer):
+    from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+    from torch.distributed.fsdp.wrap import transformer_auto_wrap_policy
+    import functools
+    assert trainer == "fsdp"
+    local_rank = int(os.getenv("LOCAL_RANK", -1))
+    from transformers.models.gpt2.modeling_gpt2 import GPT2Block
+    fsdp_model = FSDP(
+        model,
+        auto_wrap_policy=functools.partial(transformer_auto_wrap_policy, transformer_layer_cls=(GPT2Block,)),
+        device_id = torch.cuda.current_device(),
+        use_orig_params=True,
+    )
+    print(fsdp_model, "local_rank", local_rank)
+    return fsdp_model
+
+def apply_fsdp_timm_VIT_large(model, trainer):
+    from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+    from torch.distributed.fsdp.wrap import transformer_auto_wrap_policy
+    import functools
+    from timm.models.vision_transformer import Block
+    assert trainer == "fsdp"
+    local_rank = int(os.getenv("LOCAL_RANK", -1))
+    fsdp_model = FSDP(
+        model,
+        auto_wrap_policy=functools.partial(transformer_auto_wrap_policy, transformer_layer_cls=(Block,)),
+        device_id = torch.cuda.current_device(),
+        use_orig_params=True,
+    )
+    print(fsdp_model, "local_rank", local_rank)
+    return fsdp_model
+
+def benchmark_fsdp(args, executor):
+    def get_backend_name(model_args):
+        if "--torchdynamo" in model_args:
+            return "torchdynamo_" + model_args[model_args.index("--torchdynamo") + 1]
+        return "eager"
```

Comment:
does this confuse between the case of `--torchdynamo eager` (using eager backend for dynamo) and actual eager?